src/compiler/checker.ts

```diff
@@ -11638,13 +11639,25 @@ namespace ts {
                             singleProp = prop;
                         }
                         else if (prop !== singleProp) {
-                            if (!propSet) {
-                                propSet = new Map<SymbolId, Symbol>();
-                                propSet.set(getSymbolId(singleProp), singleProp);
+                            const isInstantiation = (getTargetSymbol(prop) || prop) === (getTargetSymbol(singleProp) || singleProp);
+                            // If the symbols are instances of one another with identical types - consider the symbols
+                            // equivalent and just use the first one, which thus allows us to avoid eliding private
+                            // members when intersecting a (this-)instantiations of a class with it's raw base or another instance
+                            if (!isInstantiation || !isPropertyIdenticalTo(singleProp, prop)) {
+                                if (!propSet) {
+                                    propSet = new Map<SymbolId, Symbol>();
+                                    propSet.set(getSymbolId(singleProp), singleProp);
+                                }
+                                const id = getSymbolId(prop);
+                                if (!propSet.has(id)) {
+                                    propSet.set(id, prop);
+                                }
                             }
-                            const id = getSymbolId(prop);
-                            if (!propSet.has(id)) {
-                                propSet.set(id, prop);
+                            else {
+                                // If we merged instantiations of a generic type, we replicate the symbol parent resetting behavior we used
+                                // to do when we recorded multiple distinct symbols so that we still get, eg, `Array<T>.length` printed

```

Comment:
Is the symbol parent resetting done anywhere else? If it's only in the past, I'd drop the reference and just describe it as it works now.

It could be `Array<string | number>.length` instead of `Array<T>.length`, right? How hard would that be?
Edit: Looking at the tests, it looks like the former is already what you get, at least for arrow types.

Reply:
Yes, we do it today - you'll see we do the same thing when we make the composited symbol just a few lines below.